test/e2e/network/networking_perf.go

```diff
@@ -160,3 +172,275 @@ var _ = SIGDescribe("Networking IPerf [Experimental] [Slow] [Feature:Networking-
+[  3] 5.00-6.00 sec  2.35 GBytes  20.2 Gbits/sec  19247/0
+[  3] 6.00-7.00 sec  2.43 GBytes  20.8 Gbits/sec  19867/0
+[  3] 7.00-8.00 sec  2.25 GBytes  19.4 Gbits/sec  18470/0
+[  3] 8.00-9.00 sec  2.52 GBytes  21.7 Gbits/sec  20683/0
+[  3] 0.00-9.97 sec  23.5 GBytes  20.2 Gbits/sec  192361/0
+
+Or in CSV format:
+
+20201210141800.884,10.244.2.24,47880,10.96.114.79,6789,3,0.0-1.0,1677852672,13422821376
+20201210141801.881,10.244.2.24,47880,10.96.114.79,6789,3,1.0-2.0,1980760064,15846080512
+20201210141802.883,10.244.2.24,47880,10.96.114.79,6789,3,2.0-3.0,1886650368,15093202944
+20201210141803.882,10.244.2.24,47880,10.96.114.79,6789,3,3.0-4.0,2035417088,16283336704
+20201210141804.879,10.244.2.24,47880,10.96.114.79,6789,3,4.0-5.0,1922957312,15383658496
+20201210141805.881,10.244.2.24,47880,10.96.114.79,6789,3,5.0-6.0,2095316992,16762535936
+20201210141806.882,10.244.2.24,47880,10.96.114.79,6789,3,6.0-7.0,1741291520,13930332160
+20201210141807.879,10.244.2.24,47880,10.96.114.79,6789,3,7.0-8.0,1862926336,14903410688
+20201210141808.878,10.244.2.24,47880,10.96.114.79,6789,3,8.0-9.0,1821245440,14569963520
+20201210141809.849,10.244.2.24,47880,10.96.114.79,6789,3,0.0-10.0,18752208896,15052492511
+*/
+// TODO remove other tests in this file if this test proves to be more stable, informative, and useful
+var _ = SIGDescribe("Networking IPerf2", func() {
+	// this test runs iperf2: one pod as a server, and a daemonset of clients
+	f := framework.NewDefaultFramework("network-perf")
+
+	ginkgo.It(fmt.Sprintf("should run iperf2"), func() {
+		nodeList, err := e2enode.GetReadySchedulableNodes(f.ClientSet)
+		framework.ExpectNoError(err)
+
+		serverPodsListOptions := metav1.ListOptions{
+			LabelSelector: "app=iperf2-server",
+		}
+
+		// Step 1: set up iperf2 server -- a single pod on any node
+		serverDeployment, err := iperf2ServerDeployment(f.ClientSet, f.Namespace.Name)
+		framework.ExpectNoError(err, "deploy iperf2 server deployment")
+
+		_, err = iperf2ServerService(f.ClientSet, f.Namespace.Name)
+		framework.ExpectNoError(err, "deploy iperf2 server service")
+
+		// Step 2: set up iperf2 client daemonset
+		//   initially, the clients don't do anything -- they simply pause until they're called
+		_, err = iperf2ClientDaemonSet(f.ClientSet, f.Namespace.Name)
+		framework.ExpectNoError(err, "deploy iperf2 client daemonset")
+
+		clientPodsListOptions := metav1.ListOptions{
+			LabelSelector: "app=iperf2-client",
+		}
+
+		// Make sure the server is ready to go
+		framework.ExpectNoError(e2edeployment.WaitForDeploymentComplete(f.ClientSet, serverDeployment), "unable to wait for server deployment")
+
+		// Make sure the clients are ready to go
+		err = wait.PollImmediate(2*time.Second, 100*time.Second, func() (done bool, err error) {
+			clientPodList, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).List(context.TODO(), clientPodsListOptions)
+			if err != nil {
+				return false, err
+			}
+			if len(clientPodList.Items) < len(nodeList.Items) {
```

Comment:
I think that there are clusters that doesn´t allow to schedule pods in all nodes, but I don´t know exactly how does it work, just something to take into account

Reply:
👍

Reply:
This node list is from a call to `e2enode.GetReadySchedulableNodes`, that should take care of that right?

Reply:
updated the variable name to `readySchedulableNodes`

Reply:
Good point, totally missed it, sorry